provider/azureProvider.js

```diff
@@ -112,16 +92,14 @@ return this.parsedBindings;
     }
   }
 
-  LoginWithServicePrincipal () {
-return new BbPromise((resolve, reject) => {
-      msRestAzure.loginWithServicePrincipalSecret(servicePrincipalClientId, servicePrincipalPassword, servicePrincipalTenantId, (error, credentials) => {
-        if (error) {
-          reject(error);
-        } else {
-          principalCredentials = credentials;
-          resolve(credentials);
-        }
-      });
+  Login() {
+    return login({ interactiveLoginHandler: (code) => {
+        this.serverless.cli.log(`Paste this code (copied to your clipboard) into the launched browser, and complete the authentication process: ${code}`);
```

Comment:
Does ServicePrincipal user created here show up in [portal](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal)? If yes, I think we should let users know about creating SP user.